grpc-protocol/src/main/java/com/linecorp/armeria/client/grpc/protocol/UnaryGrpcClient.java

```diff
@@ -72,11 +82,24 @@
     // cannot currently implement this without rethinking / refactoring core and punt for now since this is an
     // advanced API.
     public UnaryGrpcClient(WebClient webClient) {
+        this(webClient, UnaryGrpcSerializationFormats.PROTO);
+    }
+
+    /**
+     * Constructs a {@link UnaryGrpcClient} for the given {@link WebClient} and {@link SerializationFormat}.
+     */
+    public UnaryGrpcClient(WebClient webClient, SerializationFormat serializationFormat) {
+        if (!SUPPORTED_SERIALIZATION_FORMATS.contains(serializationFormat)) {
+            throw new IllegalArgumentException("serializationFormat : " + serializationFormat +
+                                               " (expected: one of " + SUPPORTED_SERIALIZATION_FORMATS + ')');
+        }
+        this.serializationFormat = serializationFormat;
         this.webClient = Clients.newDerivedClient(
                 webClient,
-                ClientOptions.DECORATION.newValue(
-                        ClientDecoration.of(GrpcFramingDecorator::new)
-                ));
+                ClientOptions.DECORATION.newValue(ClientDecoration
+                                                          .of((delegate, ctx, req) -> new GrpcFramingDecorator(
+                                                                  delegate, serializationFormat)
+                                                                  .execute(ctx, req))));
```

Comment:
This is probably what you actually wanted to do:
```suggestion
                ClientOptions.DECORATION.newValue(ClientDecoration.of(
                        delegate -> new GrpcGrameDecorator(delegate, serializationFormat)));
```